Azure/sonic-buildimage
# [Mellanox]New platform api -- chassis part

**- What I did**
Implement new platform API, chassis part.

**- How I did it**
make use of:
1. common/private tools available on linux to retrieve components' versions.
2. hw-management to retrieve reboot cause and eeprom information.

**- How to verify it**
use a test command to drive chassis class in order to check whether it works correctly.

**- Description for the changelog**
[Mellanox]New platform api -- chassis part

**- A picture of a cute animal (not mandatory but encouraged)**
